ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -209,34 +226,84 @@ const Footer = () => {
+  //   );
+  const isTransactionConfirmation = isCorrectDeveloperTransactionType(
+    currentConfirmation?.type,
+  );
+  const isShieldExcludedTransaction = SHIELD_EXCLUDED_TX_TYPES.includes(
+    currentConfirmation?.type as TransactionType,
+  );
+  const showShieldFooterCoverageIndicator =
+    hasUserSubscribedToShield &&
+    isTransactionConfirmation &&
+    !isShieldExcludedTransaction;
 
   return (
-    <PageFooter
-      className="confirm-footer_page-footer"
-      flexDirection={FlexDirection.Column}
-    >
-      <OriginThrottleModal
-        isOpen={showOriginThrottleModal}
-        onConfirmationCancel={onCancel}
-      />
-      <Box display={Display.Flex} flexDirection={FlexDirection.Row} gap={4}>
-        <Button
-          block
-          data-testid="confirm-footer-cancel-button"
-          onClick={handleFooterCancel}
-          size={ButtonSize.Lg}
-          variant={ButtonVariant.Secondary}
-        >
-          {t('cancel')}
-        </Button>
-        <ConfirmButton
-          alertOwnerId={currentConfirmation?.id}
-          onSubmit={() => onSubmit()}
-          disabled={isConfirmDisabled}
-          onCancel={onCancel}
+    <>
+      {showShieldFooterCoverageIndicator && <ShieldFooterCoverageIndicator />}
+      <PageFooter
+        className="confirm-footer_page-footer"
+        flexDirection={FlexDirection.Column}
+      >
+        <OriginThrottleModal
+          isOpen={showOriginThrottleModal}
+          onConfirmationCancel={onCancel}
         />
-      </Box>
-    </PageFooter>
+        <Box display={Display.Flex} flexDirection={FlexDirection.Row} gap={4}>
+          {currentConfirmation?.type !==
+            TransactionType.shieldSubscriptionApprove && (
+            <Button
+              block
+              data-testid="confirm-footer-cancel-button"
+              onClick={handleFooterCancel}
+              size={ButtonSize.Lg}
+              variant={ButtonVariant.Secondary}
+            >
+              {t('cancel')}
+            </Button>
+          )}
+          <ConfirmButton
+            alertOwnerId={currentConfirmation?.id}
+            onSubmit={onSubmit}
+            disabled={isConfirmDisabled}
+            onCancel={onCancel}
+          />
+        </Box>
+        {currentConfirmation?.type ===
+          TransactionType.shieldSubscriptionApprove && (
+          <Box display={Display.Flex} flexDirection={FlexDirection.Row} gap={4}>
+            <Text
+              variant={TextVariant.bodySm}
+              color={TextColor.textAlternative}
+            >
+              {t('shieldFooterAgreement', [
+                <a
+                  href={SHIELD_TERMS_OF_USE_URL}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  key="shield-footer-agreement-button"
+                >
+                  {t('snapsTermsOfUse')}
+                </a>,
+              ])}
+            </Text>
+          </Box>
+        )}
+      </PageFooter>
+    </>
```

Comment:
It might be useful to extract this out into another component as it is growing long.

Reply:
The changes were adding business logic to determine whether to use the coverage indicator, so I extracted that to useEnableShieldCoverageChecks.